src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -564,6 +564,15 @@ module Crystal
             if default_value = member.default_value
               counter = interpret_enum_value(default_value, enum_base_type)
             end
+
+            if default_value.is_a?(Crystal::NumberLiteral)
+              enum_base_kind = enum_base_type.kind
+              node_kind = return_enum_kind(default_value)
+              if (enum_base_kind == :i32) && (enum_base_kind != node_kind)
```

Comment:
I tried doing this, but at compile time, the compiler expects `default_value` as a Crystal::ASTNode and `#kind` is specific to Crystal::NumberLiteral, so can't call it here, 

I created `return_enum_kind` with signature expecting `(node : NumberLiteral)` then I can call `#kind` inside.

I don't know if I can handle another way, if you can point me to a solution I change immediately